### ISSUE TITLE
Function with enum argument causes error

### ISSUE THREAD

**Issue by murrant** [mockery/mockery#1198]
Calling the function on the mock results in an unexpected Undefined Constant error.
```php
enum SimpleEnum {
    case single;
}
class EnumTest {
    public function call(SimpleEnum $enum = SimpleEnum::single) {
        echo $enum->name;
    }
}

$mock = Mockery::mock('EnumTest');
$mock->shouldReceive('call')->once();
$log = $mock->call();
```

```
   Error 

  Undefined constant "SimpleEnum"

  at vendor/mockery/mockery/library/Mockery/Loader/EvalLoader.php:34
     30▕         if (class_exists($definition->getClassName(), false)) {
     31▕             return;
     32▕         }
     33▕ 
  ➜  34▕         eval("?>" . $definition->getCode());
     35▕     }
     36▕ }
     37▕ 

  1   test.php:29
      Mockery_0_EnumTest::call()
```

Calling normally works as expected.
```php
(new EnumTest())->call();
```

```
single
```

**Comment by murrant** [comment]
Looking a little more this is the errored code:
```php
public function call(\SimpleEnum $enum = SimpleEnum){
```

It is missing `::single`

**Comment by ghostwriter** [comment]
Resolved via #1199
